builder/default.nix

```diff
@@ -45,6 +45,7 @@ let
   hoogleLocal = let
     nixpkgsHoogle = import (pkgs.path + /pkgs/development/haskell-modules/hoogle.nix);
   in { packages ? [], hoogle ? pkgs.buildPackages.haskell-nix.tool "hoogle" {
+        compiler-nix-name = compiler.nix-name;
         version = "5.0.17.15";
         index-state = pkgs.haskell-nix.internalHackageIndexState;
```

Comment:
This is also a little suspicious. Could we insist that every `project` has an `index-state`? For cabal projects we have one, but I'm not sure what we'd do for stack projects (I actually don't know where we pick the index-state from for stack projects...). If we did that, then at least there's only *one* place where we might pick an index-state for the user.

(This would probably be for another PR, though)

Reply:
Definitely another PR, but I wonder if we should have a lower limit in here still, so that we know the version we ask for is available.  It might be even nicer if we could put `version = "latest"` and pick the version based on the index-state instead.

Reply:
Although, now that I think about it, arguably `hoogle` is like `hpack` etc. here, in that it's a tool that we provide, and so it's reasonable for us to pick the version (and therefore for it to change when you upgrade haskell.nix etc.). So maybe this is fine.